pandas/tests/frame/test_api.py

```diff
@@ -214,6 +214,18 @@ def test_iterrows(self):
             exp = self.mixed_frame.loc[k]
             self._assert_series_equal(v, exp)
 
+        s = self.klass(
```

Comment:
make this a separate test.

change logic to be
```
if isinstance(self.klass, SparseDataFrame):
    pytest.xfail("....give a nice message here")
```

Reply:
you *might* also be able to use the decorator form (which is preferred),e .g.

``@pytest.mark.xfail(isinstance(klass, SparseDataFrame), reason=''.....``

Reply:
I've tried xfail decorator but because this test on SparseDataFrame is inherited from SharedWithSparse, not sure how to specify the boolean condition in tests/frame/test_api.py unless it's located in tests/sparse/frame/test_frame.py where klass is declared as SparseDataFrame.

no problem using xfail imperatively inside test.